kmod/src/inode.c

```diff
@@ -1417,7 +1418,23 @@ static void init_orphan_key(struct scoutfs_key *key, u64 rid, u64 ino)
 	};
 }
 
-static int remove_orphan_item(struct super_block *sb, u64 ino)
+int scoutfs_orphan_dirty(struct super_block *sb, u64 ino)
+{
+	struct scoutfs_sb_info *sbi = SCOUTFS_SB(sb);
+	struct scoutfs_lock *lock = sbi->rid_lock;
+	struct scoutfs_key key;
+	int ret;
+
+	init_orphan_key(&key, sbi->rid, ino);
+
+	ret = scoutfs_item_dirty(sb, &key, lock);
+	if (ret == -ENOENT)
```

Comment:
Ah, that's why it wasn't returning -ENOENT.  That's not OK :).   It really needs to dirty or return an error.

Reply:
oooooooooops!